core/blockchain_ext_test.go

```
@@ -166,27 +166,42 @@ func copyMemDB(db ethdb.Database) (ethdb.Database, error) {
 	return newDB, nil
 }
 
-// This copies all files from a flat directory [src] to a new temporary directory and returns
-// the path to the new directory.
-func copyFlatDir(t *testing.T, src string) string {
+// copyDir recursively copies all files and folders from a directory [src] to a
+// new temporary directory and returns the path to the new directory.
+func copyDir(t *testing.T, src string) string {
 	t.Helper()
+
+	require := require.New(t)
 	if src == "" {
 		return ""
 	}
 
 	dst := t.TempDir()
-	ents, err := os.ReadDir(src)
-	require.NoError(t, err)
+	err := filepath.Walk(src, func(path string, info os.FileInfo, err error) error {
+		require.NoError(err)
 
-	for _, e := range ents {
-		require.False(t, e.IsDir(), "expected flat directory")
-		name := e.Name()
-		data, err := os.ReadFile(filepath.Join(src, name))
-		require.NoError(t, err)
-		info, err := e.Info()
-		require.NoError(t, err)
-		require.NoError(t, os.WriteFile(filepath.Join(dst, name), data, info.Mode().Perm()))
-	}
+		// Calculate the relative path from src
+		relPath, err := filepath.Rel(src, path)
+		require.NoError(err)
```

Why not return the error upward? I don't really know how `filepath.Walk` works, but I can look into it too